cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -176,6 +177,12 @@ protected void startUp() throws Exception {
                                           AppMetadataStore.create(context).getApplicationCount());
     metricsCollectionService.getContext(Collections.emptyMap()).gauge(Constants.Metrics.Program.APPLICATION_COUNT,
                                                                       count);
+
+    long namespaceCount = TransactionRunners.run(transactionRunner,
```

Comment:
Same comment as Terence, you can combine this transaction with the transaction on line 175 which gets the application count

Reply:
Done.